File: typescript/v1/blockytime-app/src/models/goal.ts
import { BaseModel } from './base';

export interface GoalModel extends BaseModel {
    type: number;
    hours?: number;
    duration_type?: number;
    attr_uid?: number;
    type_uid?: number;
    project_uid?: number;
    start_date?: number;
    end_date?: number;
    comment?: string;
    remind_policy?: number;
    state?: number;
    fav?: boolean;
    priority?: number;
    ext_i?: number;
    ext_t?: string;
} 

File: python/blockytime/models/goal.py
from sqlalchemy import Integer, Float, Text, Boolean, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column
from .base import Base

class Goal(Base):
    __tablename__ = 'Goal'

    uid: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    type: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    hours: Mapped[float] = mapped_column(Float)
    duration_type: Mapped[int] = mapped_column(Integer)
    attr_uid: Mapped[int] = mapped_column(Integer)
    type_uid: Mapped[int] = mapped_column(Integer, ForeignKey('Type.uid'))
    project_uid: Mapped[int] = mapped_column(Integer, ForeignKey('Project.uid'))
    start_date: Mapped[int] = mapped_column(Integer)
    end_date: Mapped[int] = mapped_column(Integer)
    comment: Mapped[str] = mapped_column(Text)
    remind_policy: Mapped[int] = mapped_column(Integer)
    state: Mapped[int] = mapped_column(Integer)
    fav: Mapped[bool] = mapped_column(Boolean)
    priority: Mapped[int] = mapped_column(Integer)
    ext_i: Mapped[int] = mapped_column(Integer)
    ext_t: Mapped[str] = mapped_column(Text) 